stdlib/fileinput.pyi

```diff
@@ -68,5 +92,12 @@ class FileInput(Iterable[AnyStr], Generic[AnyStr]):
     def isfirstline(self) -> bool: ...
     def isstdin(self) -> bool: ...
 
-def hook_compressed(filename: StrOrBytesPath, mode: str) -> IO[Any]: ...
+if sys.version_info >= (3, 10):
+    def hook_compressed(
+        filename: StrOrBytesPath, mode: str, encoding: Optional[str] = ..., errors: Optional[str] = ...
```

Comment:
Missing the `*` to mark keyword only arguments. Also, we can now use PEP 604 unions.

```suggestion
        filename: StrOrBytesPath, *, mode: str, encoding: str | None = ..., errors: str | None = ...
```